src/gc/stringobjhashtable.h

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _STRINGHASHTABLE_H_
+#define _STRINGHASHTABLE_H_
+#include "common.h"
+
+#ifdef PLATFORM_UNIX
+
+typedef char TCHAR;
+
+#else
+
+#ifndef _INC_WINDOWS
+typedef wchar_t TCHAR;
```

Comment:
That copied from `gcenv.structs.h`, which is included in `gcenv.h` and it's ok to use `gcenv.h` in .cpp's but not in headers :rage1: 

`typedef char TCHAR` for `PLATFORM_UNIX` is super confusing. Isn't `sizeof(char) == 1` _everywhere_ according to standard?
Personally I'd like to use `char16_t`, but it raising type errors when interoping with `StringObject` methods which are `WCHAR` typed.

Reply:
`TCHAR` is Windowsism: https://msdn.microsoft.com/en-us/library/office/cc842072.aspx